src/braille/internal/braille.h

```diff
@@ -0,0 +1,198 @@
+
+#include <QIODevice>
+
+#include "engraving/types/types.h"
+#include "engraving/libmscore/types.h"
+
+namespace mu::engraving {
+class Arpeggio;
+class Articulation;
+class BarLine;
+class Breath;
+class Chord;
+class ChordRest;
+class Clef;
+class DurationElement;
+class Dynamic;
+class EngravingItem;
+class Fermata;
+class Fingering;
+class Hairpin;
+class Jump;
+class KeySig;
+class Lyrics;
+class Marker;
+class Measure;
+class MeasureRepeat;
+class MMRest;
+class Note;
+class Rest;
+class Score;
+class Slur;
+class TempoText;
+class TimeSig;
+class Tuplet;
+class Volta;
+
+class BrailleEngravingItems
+{
+public:
+    BrailleEngravingItems();
+    ~BrailleEngravingItems();
+
+    void clear();
+
+    void join(BrailleEngravingItems*, bool newline = true, bool del = true);
+    void join(std::vector<BrailleEngravingItems*>, bool newline = true, bool del = true);
+
+    QString brailleStr();
+    std::vector<std::pair<EngravingItem*, std::pair<int, int> > >* items();
+
+    void setBrailleStr(QString str);
+    void addPrefixStr(QString str);
+
+    void addEngravingItem(EngravingItem*, QString braille);
+    void addLyricsItem(Lyrics*);
+
+    bool isEmpty() { return braille_str.isEmpty(); }
+    EngravingItem* getEngravingItem(int pos);
+    std::pair<int, int> getBraillePos(EngravingItem* e);
+
+    void log();
+private:
+    QString braille_str;
+    std::vector<std::pair<EngravingItem*, std::pair<int, int> > > _items;
+};
+
+//This class currently supports just a limited conversion from text to braille
+//TODO: enhance it to have full support from text to UEB, including contractions
+//http://www.brailleauthority.org/learn/braillebasic.pdf
+//https://www.teachingvisuallyimpaired.com/uploads/1/4/1/2/14122361/ueb_braille_chart.pdf
+class TextToUEBBraille
+{
+public:
+    TextToUEBBraille();
+    QString braille(QChar c);
+    QString braille(QString text);
+
+private:
+    const QString ASCII_PREFIX_CAPITAL_LETTER   = QString(",");
+    const QString ASCII_PREFIX_CAPITAL_WORD     = QString(",,");
+    const QString ASCII_PREFIX_NUMBER           = QString("#");
+    const QString ASCII_END_OF_CAPITALISATION   = QString(",");
+    const QString ASCII_END_OF_NUMBER           = QString(";");
+    QMap<QString, QString> textToBrailleASCII;
+};
+
+// Braille export is implemented according to Music Braille Code 2015
+// published by the Braille Authority of North America
+// http://www.brailleauthority.org/music/Music_Braille_Code_2015.pdf
+// This class is not thread safe.
+class Braille
+{
+public:
+    Braille(Score* s);
+    bool write(QIODevice& device);
+    bool convertMeasure(Measure* m, BrailleEngravingItems* beis);
+    bool convertItem(EngravingItem* el, BrailleEngravingItems* beis);
+
+private:
+    static constexpr int MAX_CHARS_PER_LINE = 40;
+
+    Score* score;
+
+    /* ------------ Context ------------ */
+    std::vector<Note*> previousNote;
+    std::vector<ClefType> currentCleffType;
+    std::vector<Key> currentKey;
```

Comment:
please use m_camelCase for members https://github.com/musescore/MuseScore/wiki/CodeGuidelines

And maybe better add struct like Context and use it for semantic clearacy